packages/@lwc/babel-plugin-component/src/decorators/wire/transform.js

```diff
@@ -127,12 +127,15 @@ module.exports = function transform(t, klass, decorators) {
         }
 
         const referenceLookup = scopedReferenceLookup(path.scope);
+        const isMemberExpression = id.isMemberExpression();
+        const isIdentifier = id.isIdentifier();
 
-        if (id.isIdentifier()) {
-            const adapterName = id.node.name;
-            const reference = referenceLookup(adapterName);
+        if (isIdentifier || isMemberExpression) {
+            const referenceName = isMemberExpression ? id.node.object.name : id.node.name;
+            const reference = referenceLookup(referenceName);
             wiredValue.adapter = {
-                name: adapterName,
+                name: referenceName,
+                expression: t.cloneDeep(id.node),
```

Comment:
@apapko we might need to change some metadata on platform compiler to allow for nested expressions. Maybe just sneding the MemberExpression "dot" (ex. Foo.Bar) as part of the name should be sufficient?